consensus/istanbul/backend/engine.go

```diff
@@ -860,20 +866,30 @@ func assembleExtra(header *types.Header, oldValSet []istanbul.ValidatorData, new
 			"addedValidators", common.ConvertToStringSlice(addedValidatorsAddresses), "removedValidators", removedValidators.Text(16))
 	}
 
-	ist := &types.IstanbulExtra{
-		AddedValidators:           addedValidatorsAddresses,
-		AddedValidatorsPublicKeys: addedValidatorsPublicKeys,
-		RemovedValidators:         removedValidators,
-		Seal:                      []byte{},
-		CommittedSeal:             []byte{},
+	extras, err := types.ExtractIstanbulExtra(header)
+	if err != nil {
```

Comment:
I have actually realized that this is the first time the header's `Extra` field is populated (I've confirmed the fact that the `else` branch never gets reached). 

As a result, it does not make sense to even try to extract any existing extras. 
This means that the name `updateValidatorSetDiff` does not make much sense, and it's better to rename it back to `assembleExtra` and maintain the previous functionality.

Reply:
`writeCommittedSeals` has been amended to also initialize the header's extra if it could not extract any extras from the existing header. 

This is done because it the first time the header's extras are accessed is in the `Prepare` function when we're writing the parent commits. 

Both functions are now consistent with each other